File: src/modifiers/renderingMode.ts
import { createModifier } from './createModifier';

export type RenderingMode = 'original' | 'template' | 'automatic';

/// SwiftUI `.renderingMode(_:)`. Image-only.
/// - `'template'` recolours the image with the current `foregroundColor`
/// - `'original'` keeps the source colours
/// - `'automatic'` (default) defers to SwiftUI's default for the asset
export function renderingMode(mode: RenderingMode = 'automatic') {
  return createModifier('renderingMode', { mode });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/RenderingMode.swift
import SwiftUI

/// SwiftUI `.renderingMode(_:)` — Image-only. `.template` recolours the
/// image with the current `foregroundStyle`; `.original` keeps source
/// colours. Passing `nil` (`"automatic"`) restores the system default.
enum RNWRenderingModeModifier {
    @MainActor
    static func register(into r: RNWImageModifierRegistry) {
        r.register("renderingMode") { image, params in
            switch params.string("mode") {
            case "template": return image.renderingMode(.template)
            case "original": return image.renderingMode(.original)
            default:         return image.renderingMode(nil)
            }
        }
    }
}
